### ISSUE TITLE
Procedure for pre-training

### ISSUE THREAD

**Issue by saichandrapandraju** [patil-suraj/question_generation#69]
Hi,
Can anyone explain how to pre-train a t5 for Question_Generation ? Like what are the features and labels for pre-training ?

**Comment by patil-suraj** [comment]
Hi @saichandrapandraju 

If you want to pre-train T5 on question generation then it will be similar to the fine-tuning process. The context and answer (if you have it) will be the input text and the question will be the label.

You can decide the format for input text i.e whether you want to highlight the answer special tokens or just simply prepend it to the context.

Although I'm a bit curious, why do you want to pre-train T5 for question generation?

**Comment by saichandrapandraju** [comment]
Thanks @patil-suraj .

Also, you're using another model for extracting answers for each sentence in passage. I used Spacy to extract entities and generated question for each entity.

Your approach is mostly focused on creating 'one word' answers and respective questions. Any thoughts of creating generic questions which depend on entire passage to answer ? 

I wanted to explore pre-training part and witness the evolution of the model for diff tasks. That's y I'm pre-training T5 for Question Generation
